lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -228,6 +228,7 @@ def update_positions(order: nil)
           if ActiveRecord::Acts::List.delayed_updates?
             delay(unique: true, in: 10.seconds)._execute_position_update!(sql)
           else
+            acts_as_list_list.lock.reorder(acts_as_list_class.primary_key).pluck(acts_as_list_class.primary_key)
```

Comment:
Should we acquire the lock within `_execute_position_update!` instead of here? We also execute that method as a delayed job in the other branch of the conditional - but the deadlock potential would be the same, right?

Reply:
Here in the foreground, we are already in an open transaction before we update the list of records—one of which (the current record with a callback) already has some changes saved and hence holding a lock out of turn from the rest of the list.

The delayed background update is a direct SQL statement and not in a transaction, so it can just update the set of records directly without opening a transaction and an explicit lock.